src/server/main.py

```diff
@@ -45,12 +78,16 @@ def receive_from_client(client_data):
     else:
         message = deserialised["INPUT_STRING"]
         if deserialised["PRINT_OUTPUT"] and deserialised["FILE_OUTPUT"]:
```

Comment:
Aren't we already printing the output in the "print_out" function? Printing here again would result in repetition.

Reply:
Printing XML and Binary format is not readable by the user. That's why we're printing the original message then printing the formatted message.